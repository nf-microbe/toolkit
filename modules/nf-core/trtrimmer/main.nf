process TRTRIMMER {
    tag "${meta.id}"
    label 'process_high'

    input:
    tuple val(meta), path(fasta)
    path tr_trimmer

    output:
    tuple val(meta), path("${prefix}.tr-trimmer.fasta.gz")  , emit: fasta
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    ${tr_trimmer} \\
        ${fasta} \\
        ${args} \\
        > ${prefix}.tr-trimmer.fasta

    gzip ${prefix}.tr-trimmer.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        trtrimmer: \$(tr-trimmer --version | sed 's/tr-trimmer //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo "" | gzip > ${prefix}.tr-trimmer.fasta.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        trtrimmer: \$(tr-trimmer --version | sed 's/tr-trimmer //')
    END_VERSIONS
    """
}
