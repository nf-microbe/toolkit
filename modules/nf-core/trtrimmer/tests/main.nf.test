nextflow_process {

    name "Test Process TRTRIMMER"
    script "../main.nf"
    process "TRTRIMMER"

    tag "modules"
    tag "modules_nfcore"
    tag "trtrimmer"

    test("fasta.gz") {
        when {
            params {
                trtrimmer_args ""
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true),
                ]
                input[1] = file("https://github.com/nf-microbe/modules/raw/refs/heads/master/bin/tr-trimmer", checkIfExists: true)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stub") {
        options "-stub"
        when {
            params {
                seqhasher_args ""
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true),
                ]
                input[1] = file("https://github.com/nf-microbe/modules/raw/refs/heads/master/bin/tr-trimmer", checkIfExists: true)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
