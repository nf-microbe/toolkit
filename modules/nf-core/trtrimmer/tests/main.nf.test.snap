{
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tr-trimmer.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,cadc74c4f7bd101e917616e4d452bb36"
                ],
                "fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tr-trimmer.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cadc74c4f7bd101e917616e4d452bb36"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-01T11:18:59.458449588"
    },
    "fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tr-trimmer.fasta.gz:md5,0855dd3c3fad7e77a820d09c90540b12"
                    ]
                ],
                "1": [
                    "versions.yml:md5,cadc74c4f7bd101e917616e4d452bb36"
                ],
                "fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tr-trimmer.fasta.gz:md5,0855dd3c3fad7e77a820d09c90540b12"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cadc74c4f7bd101e917616e4d452bb36"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-01T11:18:42.365088502"
    }
}